avro_schema/utils.lua

```diff
@@ -53,8 +53,137 @@ function has_only(t, key)
     return fst_key == key and snd_key == nil
 end
 
+local function map(func, tab)
```

Comment:
Why don't use `fun`?

Reply:
I have tried. 
My implementation based on the `fun` was more complex and less readable.
Possibly I do not know all of the features of `fun`. However, it seems like it is one extra reason to not use the library (it decreases readability).

Reply:
Its `map` is not the same? Ok, nevermind.